src/effects/native/reverbeffect.cpp

```diff
@@ -74,48 +97,21 @@ void ReverbEffect::processChannel(const ChannelHandle& handle,
     Q_UNUSED(handle);
     Q_UNUSED(enableState);
     Q_UNUSED(groupFeatures);
-    Q_UNUSED(sampleRate);
-    CSAMPLE bandwidth = m_pBandWidthParameter->value();
-    CSAMPLE damping = m_pDampingParameter->value();
-
-    // Flip value around.  Assumes max allowable is 1.0.
-    damping = 1.0 - damping;
-
-    bool params_changed = (damping != pState->prev_damping ||
-                           bandwidth != pState->prev_bandwidth);
-
-    pState->reverb.setBandwidth(bandwidth);
-    pState->reverb.setDecay(damping);
-
-    for (uint i = 0; i + 1 < numSamples; i += 2) {
-        CSAMPLE mono_sample = (pInput[i] + pInput[i + 1]) / 2;
-        CSAMPLE xl, xr;
 
-        // sample_t is typedefed to be the same as CSAMPLE, so no cast needed.
-        pState->reverb.process(mono_sample, damping, &xl, &xr);
-
-        pOutput[i] = xl;
-        pOutput[i + 1] = xr;
+    if (!pState || !m_pDecayParameter || !m_pBandWidthParameter || !m_pDampingParameter || !m_pSendParameter) {
+        qWarning() << "Could not retrieve all effect parameters";
+        return;
     }
 
-    if (params_changed) {
-        pState->reverb.setBandwidth(pState->prev_bandwidth);
-        pState->reverb.setDecay(pState->prev_damping);
-
-        for (uint i = 0; i + 1 < numSamples; i += 2) {
-            CSAMPLE mono_sample = (pInput[i] + pInput[i + 1]) / 2;
-            CSAMPLE xl, xr;
-
-            pState->reverb.process(mono_sample, pState->prev_damping, &xl, &xr);
-
-            pState->crossfade_buffer[i] = xl;
-            pState->crossfade_buffer[i + 1] = xr;
-        }
-
-        pState->prev_bandwidth = bandwidth;
-        pState->prev_damping = damping;
+    const auto decay = m_pDecayParameter->value();
+    const auto bandwidth = m_pBandWidthParameter->value();
+    const auto damping = m_pDampingParameter->value();
+    const auto send = m_pSendParameter->value();
 
-        SampleUtil::linearCrossfadeBuffers(
-                pOutput, pOutput, pState->crossfade_buffer, numSamples);
+    if (pState->sampleRate != sampleRate) {
+        // update the sample rate if it's changed
+        pState->reverb.init(sampleRate);
```

Comment:
This seems to be responsible for the segfault. Setting Mixxx to 96 kHz it does not segfault.